src/Region/RegionHandler.cc

```diff
@@ -2606,19 +2615,27 @@ casacore::Vector<float> RegionHandler::GetTemporaryRegionProfile(int region_idx,
             });
     } else {
         // Use BasicStats to get num_pixels and mean for current channel and stokes
-        // Get region
+        // Get region as LCRegion
         StokesRegion stokes_region;
         std::shared_ptr<casacore::LCRegion> lc_region;
         std::shared_lock frame_lock(_frames.at(file_id)->GetActiveTaskMutex());
         if (ApplyRegionToFile(region_id, file_id, z_range, stokes_index, lc_region, stokes_region)) {
-            // Get region data (report_performance = false, too much output)
+            // Get region data by applying LCRegion to image
             std::vector<float> region_data;
             if (_frames.at(file_id)->GetRegionData(stokes_region, region_data, false)) {
-                // Get BasicStats
-                BasicStats<float> basic_stats;
-                CalcBasicStats(basic_stats, region_data.data(), region_data.size());
-                num_pixels = basic_stats.num_pixels;
-                profile[0] = basic_stats.mean;
+                // Very small region, just calc needed stats here
+                num_pixels = 0;
+                double sum(0.0);
```

Comment:
I am just wondering does the performance can be further improved by adding `#pragma omp parallel for reduction(+ : sum, num_pixels)` before this for loop (?)

Reply:
Each box region along the line is very small, 1 x line_width px (max 10).  So I thought the overhead would not be worth it, but I could try it.

Reply:
I added `pragma omp parallel for` and tested a 30000-px line in the "pancake" FITS image with line widths of 3, 10, and 20.  The averages after 5 runs each (parallel vs not) showed parallel runs were 30 ms slower (width 3), 9 ms faster (width 10), and 47 ms slower (width 20).  Then I ran 5 more runs and averaged all 10; parallel runs were 5 ms faster (width 3), 42 ms slower (width 10), and 124 ms slower (width 20).  So it looks like overall the losses outweigh the gains with omp parallelization.

Reply:
Default width is 3

Reply:
yeah, I'd really expect it to be slower due to thread overhead. This is a very tight loop that is probably memory bandwidth bound more than anything else.